src/master-playlist-controller.js

```diff
@@ -871,14 +877,53 @@ export class MasterPlaylistController extends videojs.EventTarget {
 
     // Select a new playlist
     const nextPlaylist = this.selectPlaylist();
+    const logFn = error.internal ? this.logger_ : videojs.log.warn;
 
-    videojs.log.warn('Problem encountered with the current playlist.' +
-                     (error.message ? ' ' + error.message : '') +
-                     ' Switching to another playlist.');
+    logFn(`${(error.internal ? 'Internal problem' : 'Problem')} encountered with the current playlist.` +
+      `${error.message ? ` ${error.message}` : '' } Switching to another playlist.`);
 
     return this.masterPlaylistLoader_.media(nextPlaylist, isFinalRendition);
   }
 
+  /**
+   * TODO:
+   * Refactor blacklisting by codec so that it happens after
+   * we download enough of a segment to know wether it is fmp4 or ts.
+   * * If it is fmp4 we check if the browser supports the codec and blacklist
```

Comment:
Should the TODO above be below this bulleted set of comments?

Reply:
Actually, maybe it is best to put it right over where we check for `.map`, and mention that we use that check as a shorthand until the larger change is made (but that it may impact TS playlists with EXT-X-MAP tags)